haproxy_status.py

```diff
@@ -18,7 +18,8 @@ def do_GET(self):
     def do_OPTIONS(self):
         return self.do_ANY()
     def do_ANY(self):
-        if postgresql.name == etcd.current_leader()["hostname"]:
+        is_leader = postgresql.name == etcd.current_leader()["hostname"]
+        if (self.path == "/" or self.path == "/master") and is_leader or self.path == "/replica" and not is_leader:
```

Comment:
I realize you're probably not done with this yet, but this reads poorly to me. Some parens around the two major clauses would make this read a lot more clearly imo.